### DISCUSSION TITLE
On demand progress bar

### DISCUSSION THREAD

**Discussion by MohammadZarifiyan** [inertiajs/inertia#1775] [Ideas]
Hello
It would be great that if i could trigger progress bar whenever i want.
For example, Sometimes i want send a POST Request to an external API. So It would be great if i could show inertia progress bar to user before calling API and hide it when API call finished.

**Comment by vitalijalbu** [comment]
You can use **onStart** **onSuccess** or **onFinish** methods from useForm()

**Reply by MohammadZarifiyan** [reply]
What do you mean? I'm not using a form. I just want to call external API which does nothing with InertiaJS.

**Reply by vitalijalbu** [reply]
i think they use axios so you can use ueForm too... just try... else you must create a redux state and use that value on app.js file to know if progress true or false...

**Comment by RobertBoes** [comment]
Inertia by default uses nprogress, you can just use that if you want. Inertia also just makes calls to nprogress.

**Reply by MohammadZarifiyan** [reply]
Thank you. But it is better to make a section for progress bar in inertia with proxy pattern design. In this case, if the Inertia js development team decides to use another library for the progress bar, there will be no problem for the programs built by inertia js.

**Reply by RobertBoes** [reply]
It's still no problem if Inertia uses something else? You can easily use something else if Inertia decides to not use nprogress anymore.

**Reply by MohammadZarifiyan** [reply]
If inertia gives access to the progress bar, the developed software will have more integrity. Because by updating inertia js, there is no need to change the progress bar in all the parts that are used.

**Reply by MohammadZarifiyan** [reply]
Example 1:
```js
this.$inertia.startProgress();
```
Example 2:
```js
import NProgress from 'nprogress';
NProgress.start();
```
In the first example, if inertia js decides to use another package instead of nprogress, inertia js users do not need to do anything special.
But in the second example, if inertia js uses another package instead of nprogress, programmers should find all the parts that use nprogress and then try to find a solution to adapt their program to the new progress bar.

**Reply by RobertBoes** [reply]
Then make a PR on how you think it should work?
The progress bar isn't even tied to Inertia in the first place, if Inertia decides to use another library it wouldn't break anything if you're using your own implementation. I feel like you're complaining a lot without actually providing a solution, while Inertia already gives you the option to build your own solution.

**Reply by RobertBoes** [reply]
> But in the second example, if inertia js uses another package instead of nprogress, programmers should find all the parts that use nprogress and then try to find a solution to adapt their program to the new progress bar.

No they don't, because you'd never need to interact with it directly, if you do that's up to yourself. And your proxy idea would also break if you're using your own implementation.
